th/lessons/basics/mix.md

```diff
@@ -0,0 +1,134 @@
+
+```elixir
+defmodule Example.Mixfile do
+  use Mix.Project
+
+  def project do
+    [app: :example,
+     version: "0.0.1",
+     elixir: "~> 1.0",
+     build_embedded: Mix.env == :prod,
+     start_permanent: Mix.env == :prod,
+     deps: deps]
+  end
+
+  def application do
+    [applications: [:logger]]
+  end
+
+  defp deps do
+    []
+  end
+end
+```
+
+ส่วนแรกที่เราจะมาดูกันคือ `project` ในส่วนนี้เราจะระบุชื่อ application ของเรา (`app`), ระบุ version (`version`), Elixir version (`elixir`), และสุดท้าย dependencies (`deps`)
+
+ส่วน `application` จะถูกใช้ตอนสร้างไฟล์ของ application ซึ่งเราจะพูดถึงถัดไป
+
+## Interactive
+
+บางทีเราก็จำเป็นที่จะต้องใช้ `iex` กับ context ของ application เรา โอ้ขอบคุณสวรรค์ Mix ช่วยให้เราทำเรื่องนี้ได้ง่าย เราสามารถเริ่ม `iex` session ใหม่ด้วย context ของเราด้วยคำสั่ง
+
+```bash
+$ iex -S mix
+```
+
+รัน `iex` ด้วยวิธีนี้จะโหลด application ของคุณและ dependencies ต่างๆเข้าไปใน runtime ปัจจุบัน
+
+## Compilation
+
+Mix นั้นฉลาดและจะ compile เมื่อไหร่ก็ตามที่คุณแก้ไขและจำเป็นต้อง compile ใหม่ แต่การ compile โปรเจคได้เองก็ยังคงเป็นสิ่งจำเป็นอยู่ ในส่วนนี้เราจะพูดถึงการ compile โปรเจคของเราเองอย่างไร และกระบวนการ compile นั้นทำเกิดอะไรขึ้นบ้าง
+
+การ compile โปรเจค Mix เราแค่ต้องรันคำสั่ง `mix compile` ใน base directory ของเรา
+
+```bash
+$ mix compile
+```
+
+โปรเจคเรายังไม่มีอะไรมากดังนั้นผลลัพธ์เลยยังไม่มีอะไรน่าตื่นเต้นเท่าไหร่และมันควรจะเสร็จสมบูรณ์ดี
+
+```bash
+Compiled lib/example.ex
+Generated example app
+```
+
+เมื่อเรา compile โปรเจค Mix จะสร้าง directory `_build` ขึ้นมาสำหรับเก็บของที่ถูก compile ถ้าเราเข้าไปดูใน `_build` เราจะเจอกับ application ของเราที่ถูก compile แล้ว: `example.app`
+
+## Managing Dependencies
+
+ตอนนี้โปรเจคของเรายังไม่มี dependencies ดังนั้นเราจะใส่ dependencies เข้าไปและ fetching มันลงมา
+
+การจะเพิ่ม dependency เข้าไป อย่างแรกเราต้องเพิ่มเข้าไปในไฟล์ `mix.exs` ของเรา ในส่วน `deps` ซึ่งลิสของ dependency ของเราจะถูกระบุอยู่ในรูปของ tuples ด้วยค่าที่จำเป็นสองค่าและค่าเสริมอีกหนึ่งค่า ซึ่งก็คือ: ชื่อของ package ในรูปของ atom, เวอร์ชั่นในรูปของ string และสุดท้ายคือ options ที่ใส่หรือไม่ใส่ก็ได้
+
+สำหรับตัวอย่างสามารถดูได้จากโปรเจคที่มี dependencies เช่น [phoenix_slim](https://github.com/doomspork/phoenix_slim):
+
+```elixir
+def deps do
+  [{:phoenix, "~> 1.1 or ~> 1.2"},
+   {:phoenix_html, "~> 2.3"},
+   {:cowboy, "~> 1.0", only: [:dev, :test]},
+   {:slime, "~> 0.14"}]
+end
+```
+
+จากที่คุณได้เห็นจาก dependencies ข้างบน `cowboy` dependency นั้นจำเป็นแค่ตอน development และ test
+
+เมื่อเราประกาศ dependencies ของเราเสร็จเราก็เหลืออีกหนึ่งขึ้นตอนคือ fetching มัน ซึ่งมันเหมือนกับ `bundle install`
+
+```bash
+$ mix deps.get
+```
+
+แค่นั้นเอง! เมื่อเราประกาศและ fetched dependencies ของโปรเจคเราแล้ว ตอนนี้เราก็พร้อมที่จะเพิ่ม dependencies เมื่อไหร่ก็ตามที่เราต้องการ
+
+## Environments
```

Comment:
Environments
> Environment

Reply:
อันนี้มันคือหัวข้ออ่ะครับ คิดว่าไม่เปลี่ยนจะดีกว่า

Reply:
I think we shouldn't touch the title of section for consistency with original content